modules/lib/src/main/resources/assets/js/app/page/region/DescriptorBasedComponent.ts

```diff
@@ -30,30 +35,24 @@ export abstract class DescriptorBasedComponent
 
     setDescriptor(descriptor: Descriptor) {
         this.setName(descriptor ? new ComponentName(descriptor.getDisplayName()) : this.getType().getDefaultName());
-        this.setDescriptorKey(descriptor?.getKey());
-    }
 
-    private setDescriptorKey(descriptorKey: DescriptorKey) {
         const oldDescriptorKeyValue = this.descriptorKey;
 
-        this.descriptorKey = descriptorKey;
+        this.descriptorKey = descriptor?.getKey();
 
         if (!ObjectHelper.equals(oldDescriptorKeyValue, this.descriptorKey)) {
-            this.setConfig(new PropertyTree(), true);
-            this.notifyComponentUpdated(new ComponentDescriptorUpdatedEvent(this.getPath(), this.descriptorKey));
-        }
-    }
-
-    setConfig(config: PropertyTree, silent?: boolean): void {
-        const oldValue: PropertyTree = this.config;
-        if (oldValue) {
-            this.config.unChanged(this.configChangedHandler);
-        }
-        this.config = config;
-        this.config.onChanged(this.configChangedHandler);
-
-        if (!silent && !ObjectHelper.equals(oldValue, config)) {
-            this.notifyComponentUpdated(new ComponentConfigUpdatedEvent(this.getPath(), config));
+            const propertyTree = new PropertyTree();
+
+            // populating config before saving with values from form view after layout
+            const layoutPromise = !!descriptor ? new FormView(FormContext.create().setFormState(new FormState(true)).build(),
```

Comment:
Function with promise inside should be declared async and finish with the promise to avoid race conditions when using it ☝️